client/lib/config/index.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * Creates config API with data applied.
+ *
+ * @param {Object} data Config data
+ * @return {Function} Config API
+ */
+const createConfig = data => {
+	/**
+	 * Retrieves property from config by `key`.
+	 * Throws an error if the requested `key` is not set in the config file.
+	 *
+	 * @param {String} key The key of the config entry.
+	 * @return {*} Value of config or error if not found.
+	 * @api public
+	 */
+	const config = key => {
+		if ( key in data ) {
+			return data[ key ];
+		}
+		throw new Error( 'config key `' + key + '` does not exist' );
```

Comment:
> We should mirror the behavior changed in #11686 to prevent accidentally crashing large parts of the Calypso runtime when individual modules crash on import.

Yes, we should keep that in mind. If #11686 gets merged first we should move this logic in here.

> if that PR merges first (which I hope it does) then I think it would be fine to move the code here but create `config` outside of `createConfig` and inject `data` as a curried parameter maybe so that we can still test `config()` easily

Do you mean having:
```js
const config = data => key => { ... };

const createConfig = data => {
    const localConfig = config( data );
   localConfig.isEnabled = ...
  return localConfig;
};
```